lib/acts_as_taggable_on/taggable/core.rb

```diff
@@ -49,8 +50,12 @@ def #{tag_type}_list
             def #{tag_type}_list=(new_tags)
               parsed_new_list = ActsAsTaggableOn.default_parser.new(new_tags).parse
 
-              if self.class.preserve_tag_order? || parsed_new_list.sort != #{tag_type}_list.sort
-                set_attribute_was('#{tag_type}_list', #{tag_type}_list)
+              if self.class.preserve_tag_order? || (parsed_new_list.sort != #{tag_type}_list.sort)
+                if ActsAsTaggableOn::Utils.legacy_activerecord?
+                  set_attribute_was("#{tag_type}_list", #{tag_type}_list)
+                else
+                  attribute_change("#{tag_type}_list")
```

Comment:
```suggestion
                  send("#{tag_type}_list_will_change!")
```

Complete shot in the dark here 😄